norman_utils_external/streaming_utils.py

```diff
@@ -24,11 +62,54 @@ async def chain_streams(*streams: Union[Iterable[bytes], AsyncIterable[bytes]]):
 
     @staticmethod
     async def process_read_stream(
-            file_stream: Union[AsyncBufferedReader, BufferedReader] ,
-            processor: Processor[T],
-            chunk_size: int,
-            yield_processed: bool = True
+        file_stream: Union[AsyncBufferedReader, BufferedReader],
+        processor: Processor[T],
+        chunk_size: int,
+        yield_processed: bool = True
     ) -> AsyncGenerator[Union[bytes, T], None]:
+        """
+        Read a file/stream in chunks and optionally process each chunk before
+        yielding it. Supports both synchronous and asynchronous `.read()`
+        methods.
+
+        This helper is useful when streaming large files or socket data and
+        applying on-the-fly transformations, such as:
+        - hashing
+        - compression / decompression
+        - encryption / decryption
+        - line parsing
+        - transcoding
+
+        **Parameters**
+
+        - **file_stream** (`AsyncBufferedReader | BufferedReader`)
+          Object with a `.read(chunk_size)` method.
+          If the method is async, it will be awaited automatically.
+
+        - **processor** (`Callable[[bytes], T] | Callable[[bytes], Any]`)
+          Function applied to each chunk.
+          - If `yield_processed=True`, the output of this function is yielded.
+          - If `yield_processed=False`, the raw chunk is yielded instead.
+
+        - **chunk_size** (`int`)
+          Maximum number of bytes to read per iteration.
+
+        - **yield_processed** (`bool`, default `True`)
+          Controls what gets yielded:
+          - `True` → yield `processor(chunk)`
+          - `False` → yield the raw `bytes` chunk
+
+        **Yields**
+
+        - If `yield_processed=True`: values of type `T` returned by `processor()`
+        - If `yield_processed=False`: raw `bytes` chunks
+
+        **Stops When**
```

Comment:
Remove unsanctioned section (internal implementation details are amenable to change)